runtime/autoload/provider/perl.vim

```diff
@@ -0,0 +1,69 @@
+if exists('s:loaded_perl_provider')
+  finish
+endif
+
+let s:loaded_perl_provider = 1
+
+function! provider#perl#Detect() abort
+  " use g:perl_host_prof if set or check if perl is on the path
+  let prog = exepath(get(g:, 'perl_host_prog', 'perl'))
+  if empty(prog)
+    return ''
+  endif
+
+  " if perl is available, make sure the required module is available
+  call system([prog, '-W', '-MNeovim::Ext', '-e', ''])
```

Comment:
Used `system` because this is quick. Could use `jobstart` for https://metacpan.org/pod/local::lib, `perlbrew`, https://github.com/tokuhirom/plenv.